go/src/koding/klient/machine/machinegroup/ssh_test.go

```diff
@@ -0,0 +1,166 @@
+package machinegroup
+
+import (
+	"io/ioutil"
+	"os"
+	"os/user"
+	"strconv"
+	"testing"
+	"time"
+
+	"koding/kites/tunnelproxy/discover/discovertest"
+	"koding/klient/machine"
+	"koding/klient/machine/client/clienttest"
+)
+
+func TestSSH(t *testing.T) {
+	var (
+		builder = clienttest.NewBuilder(nil)
+		id      = machine.ID("serv")
+	)
+
+	wd, err := ioutil.TempDir("", "ssh")
+	if err != nil {
+		t.Fatalf("want err = nil; got %v", err)
+	}
+	defer os.RemoveAll(wd)
+
+	g, err := New(testOptions(wd, builder))
+	if err != nil {
+		t.Fatalf("want err = nil; got %v", err)
+	}
+	defer g.Close()
+
+	// Create test machine with IP address.
+	const fakeHost = "127.0.32.123"
+	createReq := &CreateRequest{
+		Addresses: map[machine.ID][]machine.Addr{
+			id: {
+				clienttest.TurnOnAddr(),
+				machine.Addr{
+					Network:   "ip",
+					Value:     fakeHost,
+					UpdatedAt: time.Now(),
+				},
+			},
+		},
+	}
+	if _, err := g.Create(createReq); err != nil {
```

Comment:
no, machine ID, is a unique index (`id` field L38). When we call `machine list`, the list of machines is sent to the klient which then calls `g.Create` using simillar `CreateRequest`.

- If called twice and machine address doesn't change, nothing will happen.
- If called twice and machine address change, new address will be added to this machine (this handles machines with dynamic IPs)

This logic is here to have a machine into which we will be SSH-ing in.